woocommerce/woocommerce-android
# Activate: Add support for internal remote feature flag

Closes: #8998 


### Description


This PR introduces the ability to use internal remote feature flags instead of relying on Firebase. The same mechanism is already used in the WordPress and Jetpack apps.

Since this work has architectural implications, I'm seeking as much feedback as possible to ensure that it is intuitive, user-friendly, and future-proof. Here are some initial considerations:

1. Use the existing `FeatureFlagStore` in FluxC.
2. Create a new enum class (`RemoteFeatureFlag`) that functions similarly to the existing `FeatureFlag` class, with a use case class `CheckRemoteFeatureFlagEnabled` for doing the checking.
3. Provide the ability to fetch and update all remote flags at once, including fetching by app version.
4. Provide the ability to get the value of a single remote flag from the database.

### Use cases:

#### Adding a new remote feature flag into the app
1. Update `RemoteFeatureFlag` enum with  a new constant containing the remote key, then
2. Update `CheckRemoteFeatureFlagEnabled` to handle the new enum constant. `PackageUtils.isDebugBuild()` can be used to force-enable a feature during development purposes, and can be removed as needed.

#### Fetching latest remote feature flags
`WPComRemoteFeatureFlagRepository.fetchFeatureFlags()` is available for this. This function behind-the-scene saves the fetched values to the DB, which is then used for checking individual remote flag value as below.

#### Checking whether a remote feature flag is enabled
Assuming `fetchFeatureFlags()` above has been called recently, inject and use the `CheckRemoteFeatureFlagEnabled(REMOTE_FEATURE_FLAG_CONSTANT)` usecase class.


### Testing instructions


For now, ensure unit tests run properly. Do further testing once this functionality is actually used.


### Images/gif
